api/src/main/java/io/druid/segment/loading/DataSegmentArchiver.java

```diff
@@ -21,8 +21,31 @@
 
 import io.druid.timeline.DataSegment;
 
+import javax.annotation.Nullable;
+
 public interface DataSegmentArchiver
 {
-  public DataSegment archive(DataSegment segment) throws SegmentLoadingException;
-  public DataSegment restore(DataSegment segment) throws SegmentLoadingException;
+  /**
+   * Perform an archive task on the segment and return the resulting segment or null if there was no action needed.
+   *
+   * @param segment The source segment
+   *
+   * @return The segment after archiving or `null` if there was no archiving performed.
+   *
+   * @throws SegmentLoadingException on error
+   */
+  @Nullable
+  DataSegment archive(DataSegment segment) throws SegmentLoadingException;
```

Comment:
Instead of returning null, perhaps a nicer API would be to return an equivalent object. Then callers don't need to deal with potentially null returns. If a caller wants to know if anything was actually done, they could check `retVal.equals(segment)` rather than `retVal != null`.